core/string.cpp

```diff
@@ -45,4 +49,30 @@ std::map<DType, std::string> &dtypeNameRegistry() {
   return registry;
 }
 
+const std::string to_iso_date(const scipp::core::time_point &item,
+                              const std::string &unit) {
```

Comment:
Maybe `units::Unit` would be better choice for the argument type? `std::string` could be anything. In the higher-level functions where the unit is optional use `std::optional<units::Unit>`?